orca-clouddriver/src/test/java/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfigurationTest.java

```diff
@@ -0,0 +1,104 @@
+import com.netflix.spinnaker.config.okhttp3.OkHttpClientProvider;
+import com.netflix.spinnaker.okhttp.SpinnakerRequestInterceptor;
+import com.netflix.spinnaker.orca.clouddriver.KatoRestService;
+import com.netflix.spinnaker.orca.test.YamlFileApplicationContextInitializer;
+import java.util.List;
+import okhttp3.OkHttpClient;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+
+@ExtendWith(MockitoExtension.class)
+@ContextConfiguration(
+    classes = CloudDriverConfigurationProperties.class,
+    initializers = CloudDriverConfigurationTest.class)
+@EnableConfigurationProperties
+@SpringBootTest
+public class CloudDriverConfigurationTest extends YamlFileApplicationContextInitializer {
+
+  @Autowired private CloudDriverConfigurationProperties cloudDriverConfigurationProperties;
+
+  private CloudDriverConfiguration.ClouddriverRetrofitBuilder clouddriverRetrofitBuilder;
+
+  @Override
+  protected String getResourceLocation() {
+    return "classpath:clouddriver-sharding-properties.yml";
+  }
+
+  @BeforeEach
+  public void setup() {
+    OkHttpClientBuilderProvider okHttpClientBuilderProvider =
+        new OkHttpClientBuilderProvider() {
+          @Override
+          @NotNull
+          public Boolean supports(@NotNull ServiceEndpoint service) {
+            return true;
+          }
+
+          @Override
+          @NotNull
+          public OkHttpClient.Builder get(@NotNull ServiceEndpoint service) {
+            return new OkHttpClient().newBuilder();
+          }
+        };
+    OkHttpClientProvider okHttpClientProvider =
+        new OkHttpClientProvider(List.of(okHttpClientBuilderProvider));
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    RestAdapter.LogLevel logLevel = RestAdapter.LogLevel.FULL;
+    RequestInterceptor requestInterceptor = new SpinnakerRequestInterceptor(null);
+
+    this.clouddriverRetrofitBuilder =
+        new CloudDriverConfiguration.ClouddriverRetrofitBuilder(
+            objectMapper,
+            okHttpClientProvider,
+            logLevel,
+            requestInterceptor,
+            cloudDriverConfigurationProperties);
+  }
+
+  @DisplayName("when selector config is a YAML list, successfully constructs a SelectableService")
+  @Test
+  public void testConstructingSelectorsWithYAMLList() {
+    /*
+     * Because the configuration of a ServiceSelector is a Map<String, Object>, Jackson converts a YAML list
```

Comment:
[ServiceSelector is an interface](https://github.com/spinnaker/kork/blob/5c70628ddaca00015222264e89d24faaa3bd9fd3/kork-web/src/main/java/com/netflix/spinnaker/kork/web/selector/ServiceSelector.java) with no mention of `Map<String, Object>`.  Maybe better to reference ServiceSelector implementations, e.g. [ByExecutionTypeServiceSelector](https://github.com/spinnaker/kork/blob/5c70628ddaca00015222264e89d24faaa3bd9fd3/kork-web/src/main/java/com/netflix/spinnaker/kork/web/selector/ByExecutionTypeServiceSelector.java#L29)?  But then also it seems like the basis of this is the type of the [config member of BaseUrl](https://github.com/spinnaker/orca/blob/d1e42deb42aae0aa9e9089a95cc2dfb298030362/orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfigurationProperties.java#L42) -- Map<String, Object>, and then that getting passed to ServiceSelector constructors [here](https://github.com/spinnaker/orca/pull/4287/files#diff-4e82bc2b7d7efc31ab7780eeb09035ed0f8993a897f66d3c040bdfbbd926f09cR185).

Reply:
Updated the comment to better reflect what's going on, and why this test is necessary. Let me know if it's any better.